openedx/features/journals/api.py

```diff
@@ -0,0 +1,276 @@
+    Determines whether the Journals app is installed and enabled
+    """
+    return 'openedx.features.journals.apps.JournalsConfig' \
+        in settings.INSTALLED_APPS and settings.FEATURES.get('ENABLE_JOURNAL_INTEGRATION', False)
+
+
+def get_journals(site):
+    """Retrieve journals from the discovery service.
+
+    Keyword Arguments:
+        site (Site): Site object for the request
+        will be returned
+
+    Returns:
+        list of dict, representing journals
+    """
+    if not journals_enabled():
+        return []
+
+    api_resource = 'journals'
+    orgs = configuration_helpers.get_current_site_orgs()
+
+    cache_key = get_cache_key(
+        site_domain=site.domain,
+        resource=api_resource,
+        orgs=orgs
+    )
+
+    # look up in cache
+    journals = cache.get(cache_key)
+
+    if not journals:
+        api_client = DiscoveryApiClient()
+        if not api_client:
+            return []
+        journals = api_client.get_journals(orgs)
+        cache.set(cache_key, journals, JOURNALS_CACHE_TIMEOUT)
+
+    return journals
+
+
+def fix_course_images(bundle):
+    """
+        Set the image for a course. If the course has an image, use that. Otherwise use the first
+        course run that has an image.
+    """
+    for course in bundle['courses']:
+        course_image = course['image'].get('src') if course.get('image') else None
+        if course_image:
+            # Course already had image and we don't need to check course runs
+            continue
+        for course_run in course['course_runs']:
+            if course_run['image']:
+                course['image'] = course_run['image']
+                break
+
+
+def get_journal_bundles(site, bundle_uuid=''):
+    """Retrieve journal bundles from the discovery service.
+
+    Returns:
+        list of dict, representing journal bundles
+    """
+    if not journals_enabled():
+        return []
+
+    api_resource = 'journal_bundles'
+
+    cache_key = get_cache_key(
+        site_domain=site.domain,
+        resource=api_resource,
+        bundle_uuid=bundle_uuid
+    )
+
+    journal_bundles = cache.get(cache_key)
+
+    if not journal_bundles:
+        api_client = DiscoveryApiClient()
+        if not api_client:
+            return []
+        journal_bundles = api_client.get_journal_bundles(uuid=bundle_uuid)
+        cache.set(cache_key, journal_bundles, JOURNALS_CACHE_TIMEOUT)
+
+    if isinstance(journal_bundles, dict):
+        fix_course_images(journal_bundles)
+    else:
+        for bundle in journal_bundles:
+            fix_course_images(bundle)
+
+    return journal_bundles
+
+
+def get_journals_root_url():
+    """
+    Return the base url used to display Journals
+    """
+    if journals_enabled():
+        return configuration_helpers.get_configuration_value(
+            'JOURNALS_URL_ROOT',
+            settings.JOURNALS_URL_ROOT
+        ) if configuration_helpers.is_site_configuration_enabled() else settings.JOURNALS_URL_ROOT
```

Comment:
NIT:  can we nest this whole conditional for readability.

Reply:
fixed